src/main/scala/net/heartsavior/spark/sql/state/StateStoreRelation.scala

```diff
@@ -41,9 +40,7 @@ class StateStoreRelation(
 
   override def sqlContext: SQLContext = session.sqlContext
 
-  override def schema: StructType = new StructType()
-    .add("key", StructType(keySchema.fields), nullable = false)
-    .add("value", StructType(valueSchema.fields), nullable = false)
+  override def schema: StructType = SchemaUtil.schema(keySchema, valueSchema)
```

Comment:
The schema information is specific to here, hence good to leave it here as it is. The method name `schema` is not clear to determine "which" schema it denotes.

Reply:
I moved it to `SchemaUtil` because I need it both in `StateStoreRelation` and `StateStoreReaderRDD`. I agree that the name `schema` is not ideal. would `keyValuePairSchema` work better for you?

Reply:
Sounds OK to me.